### DISCUSSION TITLE
Using the Apache Port without using Reverse proxy mode

### DISCUSSION THREAD

**Discussion by Leahnessa** [nextcloud/all-in-one#4974] [Questions] [answered]
# Problem
When I try to use a custom port for the Apache Container using the `--env APACHE_PORT=11000` as described in the [reverse proxy documentation](https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md), to bind the interface to a different port, and then connect to the container via my NGINX Reverse Proxy I get the error: `SSL_ERROR_RX_RECORD_TOO_LONG`. All connections to the other application as well as the direct connection to the server via HTTP (not s) do work. This leads me to believe that it might have something todo with the lack of encryption in reverse proxy mode, as I only work with the encrypted packets, and therefore don't do any conversion to HTTPS. (Though that is only a guess since I honestly have no idea). The connection also works if I disable NGINX and let the container run normally.
# The Question
So the Question is, if its possible to run the server on a custom port without it disableing the internal proxy, as i want to keep my 443 port free for later.

This is my NGINX configuration
```
user nginx;
worker_processes auto;
error_log /var/log/nginx/error.log notice;
pid /run/nginx.pid;


# Load dynamic modules. See /usr/share/doc/nginx/README.dynamic.
include /usr/share/nginx/modules/*.conf;

events {
    worker_connections 1024;
}

stream{

    upstream admin {
        server localhost:9090;  
    }

    upstream nextcloud {
        server localhost:11000;
    }

    map $ssl_preread_server_name $upstream{
        hostnames;
        default         nextcloud;
        admin.*         admin;
        nextcloud.*     nextcloud;      

   }
   server {
        listen 443;
        ssl_preread on;
        proxy_pass $upstream;
    }
}
```
And this the run configuration:
```
# For Linux and without a web server or reverse proxy (like Apache, Nginx, Cloudflare Tunnel and else) already in place:
sudo docker run \
--init \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 5050:80 \
--publish 7070:8080 \
--publish 8443:8443 \
--env APACHE_PORT=11000 \
--env APACHE_IP_BINDING=127.0.0.1 \
--env SKIP_DOMAIN_VALIDATION=true \
--volume nextcloud_aio_mastercontainer:/mnt/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

**Comment by szaimen** [comment, marked as answer]
Hi, can you follow https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md#6-how-to-debug-things?

**Reply by Leahnessa** [reply]
thank you for the reply, so I take it, there is no way to set a custom port, and keep the container in normal mode and running another proxy just for encrypting the traffic would be needed, right? (like an environment variable that tells it to be in normal mode.)

**Reply by szaimen** [reply]
yes indeed
